lib/src/idx/ft/postings.rs

```diff
@@ -144,5 +144,6 @@ mod tests {
 
 		// The underlying b-tree should be empty now
 		assert_eq!(p.statistics(&mut tx).await.unwrap().keys_count, 0);
+		tx.commit().await.unwrap();
```

Comment:
@phughk @sgirones Any idea why we need to commit not only write transactions but also read-only transactions like this one? I thought a read never blocks writes in TiKV and therefore read only transactions never conflict. Correct me if I'm missing anything 🙏

Reply:
You are right; This was a search replace 😂 Would like to "end" the transaction just in case - it won't hurt to commit a read transaction and it prevents problems if it would get refactored into a write transaction.

Reply:
> it prevents problems if it would get refactored into a write transaction.

That's a great point @phughk! Then this looks good!

Reply:
Yeah actually commit fails because it isn't write.
I have changed it to cancel instead.
We do want ALL transactions to resolve, even though for reads that isn't important. The reason is that in production we will have warnings for all unclosed transactions. So resolving read transactions is important.

Even though there is a timeout, we don't want the timeout to impact snapshotting, log compaction, and garbage collection on storage